common-npm-packages/azure-arm-rest/azure-arm-common.ts

```diff
@@ -516,9 +516,9 @@ export class ApplicationTokenCredentials {
                 tl.debug(`MSAL - retrying getMSALToken - temporary error code: ${error.errorCode}`);
                 tl.debug(`MSAL - retrying getMSALToken - remaining attempts: ${retryCount}`);
 
-                // Wait for a backoff time before retrying
-                await new Promise(r => setTimeout(r, Math.min(retryWaitMS * retryCount, MAX_CREATE_AAD_TOKEN_BACKOFF_TIMEOUT)));
-                return await this.getMSALToken(force, (retryCount - 1), retryWaitMS);
+                // Wait for exponential backoff time before retrying (2^attempt * base wait time)
+                await new Promise(r => setTimeout(r, Math.min(retryWaitMS * Math.pow(2, exponentialBackoffTryCount), MAX_CREATE_AAD_TOKEN_BACKOFF_TIMEOUT)));
+                return await this.getMSALToken(force, (retryCount - 1), retryWaitMS, exponentialBackoffTryCount + 1);
```

Comment:
Thanks for update. I think it is not even necessary to introduce `exponentialBackoffTryCount` parameter. You can just do following:
```ts
await new Promise(r => setTimeout(r, Math.min(retryWaitMS, MAX_CREATE_AAD_TOKEN_BACKOFF_TIMEOUT)));
return await this.getMSALToken(force, (retryCount - 1), retryWaitMS * 2);
```